### DISCUSSION TITLE
toolbar missing

### DISCUSSION THREAD

**Discussion by albgen** [rustdesk/rustdesk#4028] [General]
![image](https://user-images.githubusercontent.com/7231968/231426441-5acd3465-ea0b-4c11-947b-45cba05d973a.png)
How can i show this?

mine is looking like this
![image](https://user-images.githubusercontent.com/7231968/231426557-721e8c44-9c5e-4f63-896b-a044ca8b99bd.png)

**Comment by rustdesk** [comment]
Nightly build: https://github.com/rustdesk/rustdesk/releases/tag/nightly

**Comment by albgen** [comment]
thanks. Just tried but i see only the legacy mode and the other options are not there

**Reply by albgen** [reply]
needed to remove old settings. Basically empty the folder  %appdata%\RustDesk

**Reply by rustdesk** [reply]
Really? @fufesou take care of this, we need to find the cause, we can not require users do this operation.

**Reply by albgen** [reply]
yes, just deleted on both clients the content of %appdata%\RustDesk and the toolbar appeared. 

Most probably it's a feature :). I mean reading old setting, the client simply stick to less options possible...

**Reply by rustdesk** [reply]
Thanks for your info. It is super helpful.

**Reply by fufesou** [reply]
@albgen The three options should be shown if both sides run the nightly build.

I've read the code, but failed to reason why only "Legacy Mode" is shown.

I've also tried the following steps:
1. Remove the `%appdata%\RustDesk` folder. Clear RustDesk data.
2. Run 1.1.9 portable on both sides.     --------------             No keyboard mode option.
3. Run the nightly build on both sides.  ------------------- Legacy Mode, Map Mode, Translate Mode Beta
4. Run the nightly build on local side and 1.1.9 on the remote side ----------------------------- Legacy Mode

I find the 4th step show only "Legacy Mode" option.

Can you please run again to find the differences?

**Reply by fufesou** [reply]
@albgen Have you ever restarted rustdesk after removing `%appdata%\RustDesk`? If yes, how did you restart RustDesk?

**Reply by albgen** [reply]
Restarting rustdesk for me it's just double-click on the icon.

> @albgen Have you ever restarted rustdesk after removing `%appdata%\RustDesk`? If yes, how did you restart RustDesk?

**Reply by albgen** [reply]
> @albgen The three options should be shown if both sides run the nightly build.
> 
> I've read the code, but failed to reason why only "Legacy Mode" is shown.
> 
> I've also tried the following steps:
> 
> 1. Remove the `%appdata%\RustDesk` folder. Clear RustDesk data.
> 2. Run 1.1.9 portable on both sides.     --------------             No keyboard mode option.
> 3. Run the nightly build on both sides.  ------------------- Legacy Mode, Map Mode, Translate Mode Beta
> 4. Run the nightly build on local side and 1.1.9 on the remote side ----------------------------- Legacy Mode
> 
> I find the 4th step show only "Legacy Mode" option.
> 
> Can you please run again to find the differences?

Did the same steps as you, and i can confirm that i see the same as per your tests...
Don't know what happened yesterday.

**Reply by rustdesk** [reply]
@albgen Thanks for your time.

**Reply by albgen** [reply]
thanks to you for this great tool!
